src/mapContainer.ts

```diff
@@ -695,6 +931,67 @@ export class MapContainer {
                     ev.stopPropagation();
                 });
         });
+        newMarker.on('move', (event: leaflet.LeafletEvent) => {
+            let oldMarkerLocation = marker.location.toString();
+            let vertices = this.display.vertices;
+            if (vertices.has(oldMarkerLocation)) {
+                let vertexToUpdate = vertices.get(oldMarkerLocation);
+                let newLatLng = newMarker.getLatLng().clone();
+                for (let edge of vertexToUpdate) {
+                    if (edge.polyline) {
+                        if (edge.v1.toString() === oldMarkerLocation) {
+                            edge.v1 = newLatLng;
+                        } else if (edge.v2.toString() === oldMarkerLocation) {
+                            edge.v2 = newLatLng;
+                        }
+                        edge.polyline.remove();
+                        this.display.polylines = this.display.polylines.filter(
+                            (x) => x !== edge.polyline
+                        );
+                        let polyline = leaflet.polyline([edge.v1, edge.v2], {
+                            color: this.settings.edgeColor,
+                            weight: 1,
+                        });
+                        edge.polyline = polyline;
+                        polyline.addTo(this.display.map);
+                        this.display.polylines.push(polyline);
+                    }
+                }
+                marker.location = newLatLng;
+                vertices.delete(oldMarkerLocation);
+                vertices.set(marker.location.toString(), vertexToUpdate);
+            }
+        });
+        newMarker.on('moveend', async (event: leaflet.LeafletEvent) => {
+            const content = await this.app.vault.read(marker.file);
```

Comment:
Since this is a low-level operation on files, I think it would be better to export it to a function in utils.ts and document.